lib/xeroizer/oauth2_factory.rb

```diff
@@ -0,0 +1,7 @@
+module Xeroizer
+   class OAuth2Factory
```

Comment:
Maybe skip the factory pattern, unless there's a lot of wiring up that happens?

Its slightly more verbose to do 
```Xeroizer::OAuth2Factory.build(x, y, z)```
than
```Xeroizer::OAuth2.new(x, y, z)```

If it had some logic to it, I could see the use.

Reply:
Some more code we need to unwind